packages/core/src/api/nodeUtil.test.ts

```diff
@@ -0,0 +1,787 @@
+import { describe, expect, it, vi, beforeEach, afterEach } from "vitest";
+
+import { setupTestEnv } from "./blockManipulation/setupTestEnv.js";
+import { getBlocksChangedByTransaction } from "./nodeUtil.js";
+import { Transaction } from "prosemirror-state";
+import { BlockNoteEditor } from "../editor/BlockNoteEditor.js";
+import { Step } from "prosemirror-transform";
+
+const getEditor = setupTestEnv();
+
+describe("Test getBlocksChangedByTransaction", () => {
+  let transaction: Transaction;
+  let editor: BlockNoteEditor;
+  let originalDispatch: typeof editor.dispatch;
+
+  beforeEach(() => {
+    transaction = undefined as unknown as Transaction;
+    editor = getEditor();
+    originalDispatch = editor.dispatch;
+    const mockDispatch = vi.fn((tr) => {
+      editor._tiptapEditor.dispatch(tr);
+      if (transaction) {
+        tr.steps.forEach((step: Step) => {
+          transaction.step(step);
+        });
+      } else {
+        transaction = tr;
+      }
+    });
+    editor.dispatch = mockDispatch;
+  });
+
+  afterEach(() => {
+    editor.dispatch = originalDispatch;
+  });
+
+  it("should return the correct blocks changed by a transaction", () => {
+    const transaction = editor.transaction;
+    const blocksChanged = getBlocksChangedByTransaction(transaction, editor);
+    expect(blocksChanged).toEqual([]);
+  });
+
+  it("should return blocks inserted by a transaction", () => {
+    editor.insertBlocks([{ type: "paragraph" }], "paragraph-0", "after");
+
+    const blocksChanged = getBlocksChangedByTransaction(transaction!, editor);
+
+    expect(blocksChanged).toEqual([
```

Comment:
Do we wanna use `toEqual` in these tests? We generally use `toMatchFileSnapshot` to verify JSON objects, and it would make this file a lot smaller

Reply:
+ the JSON to compare with is generated automatically so writing the tests is a bit less cumbersome

Reply:
Habit, but, yea I guess that is slightly better.